src/core.rs

```diff
@@ -96,7 +97,7 @@ impl BPF {
             let size = bpf_function_size(self.p, cname.as_ptr()) as i32;
             let license = bpf_module_license(self.p);
             let version = bpf_module_kern_version(self.p);
-            if start == 0 as *mut bpf_insn {
+            if start == ptr::null_mut() {
```

Comment:
These comparisons would be better as [`<ptr>.is_null()`](https://doc.rust-lang.org/std/primitive.pointer.html#method.is_null)